vuln-reach/src/javascript/package/reachability.rs

```diff
@@ -791,4 +792,51 @@ mod tests {
         println!("{r:#?}");
         println!("{r}");
     }
+
+    // This test case's main purpose is checking whether imports without extension
+    // (i.e. './lib/Pool', like generic_pool does) are still correctly resolved.
+    #[tokio::test]
+    async fn test_reachability_generic_pool() {
+        let package = npm_package("generic-pool", "3.8.2").await;
+        let r = PackageReachability::new(
+            &package,
+            &VulnerableNode::new("generic-pool", "lib/Pool.js".to_string(), 23, 6, 23, 9),
+        )
+        .unwrap();
+
+        println!("{r:#?}");
+        println!("{r}");
+
+        let node_paths_in_index = r.0.get("index.js").unwrap().get("lib/Pool.js").unwrap();
+
+        assert!(node_paths_in_index.iter().map(|node_path| node_path.0.first().unwrap()).any(
+            |node_step| node_step.symbol == "createPool"
+                && node_step.start_row == 9
+                && node_step.start_column == 40
+        ));
+    }
+
+    #[tokio::test]
+    async fn test_reachability_handlebars_esm() {
+        let package = npm_package("handlebars", "4.7.6").await;
+
+        let target_node =
+            VulnerableNode::new("handlebars", "dist/cjs/handlebars/runtime.js", 51, 9, 51, 16);
+        let r = PackageReachability::new(&package, &target_node).unwrap();
+
+        println!("{r:#?}");
+        println!("{r}");
+    }
```

Comment:
Why no asserts here?

Reply:
At the moment the test failure is led by the unwrap above these. I don't think there is any behavior to be tested at this spot for which the result is not a good proxy anyway.